### PR TITLE
feat: POS multicurrency support

### PR DESCRIPTION
**Pull request by this-gavagai**
This PR provides multicurrency functionality to the POS.

At present, the POS does not provide any mechanisms for managing multiple currencies, and furthermore it tolerates incoherent data if a user attempts to use conflicting modes of payment within a POS profile. This PR adds logic, validation, and user interface elements to allow multicurrency functionality while making as few changes as possible to the underlying architecture.

Significant code changes are as follows:
- Currency fields added to both POS Opening Entry and POS Closing Entry
- In POS Closing Entry, summary fields for total sales made consistent to always display in company default currency
- Some verification checks added to ensure that account currencies match POS Invoice currency
- Extended POS Invoice Merge Log document to separate consolidated invoices by currency
- Added UI methods to change currency via menu item
- Filter available payment modes to only those matching POS Invoice currency

### CODE REVIEW

**Comment by this-gavagai** [comment]
Hello, is there anything I can do to facilitate code review?

**Comment by this-gavagai** [comment]
Am I misunderstanding the process for feature submissions? It's almost three weeks now without even acknowledgment of this PR. Any suggestions would be greatly appreciated. @nextchamp-saqib

**Comment by nextchamp-saqib** [comment]
Hey, @this-gavagai

I have gone through the changes, but haven't found time to review/pull & test locally.
I have kept this as well as other POS features on my to-do list. I'll review this soon. 

In the meantime, you could add test cases for simple multi currency scenarios.

Thanks for contributing!

File: erpnext/accounts/doctype/pos_invoice/pos_invoice.py [hunk 1]
```diff
@@ -377,7 +380,7 @@ def set_pos_fields(self, for_validate=False):
 				)
 				customer_group_price_list = frappe.db.get_value("Customer Group", customer_group, 'default_price_list')
 				selling_price_list = customer_price_list or customer_group_price_list or profile.get('selling_price_list')
-				if customer_currency != profile.get('currency'):
+				if customer_currency and customer_currency != profile.get('currency') and not self.currency:
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/pos_invoice/pos_invoice.py:383]:
```suggestion
				if customer_currency and customer_currency != profile.get('currency'):
```

Customer Currency should overwrite existing currency in the Invoice, right?
Is this intentional? If yes, Can you help me understand with an example?

File: erpnext/accounts/doctype/pos_invoice/pos_invoice.py [hunk 2]
```diff
@@ -269,6 +268,10 @@ def validate_change_account(self):
 			frappe.db.get_value("Account", self.account_for_change_amount, "company") != self.company:
 			frappe.throw(_("The selected change account {} doesn't belongs to Company {}.").format(self.account_for_change_amount, self.company))
 
+		# verify that change account matches invoice currency
+		if self.change_amount and self.currency != frappe.db.get_value("Account", self.account_for_change_amount, "account_currency"):
+			frappe.throw(_("Change Account does not match invoice currency"))
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/pos_invoice/pos_invoice.py:273]:
```suggestion
		change_account_currency = frappe.db.get_value("Account", self.account_for_change_amount, "account_currency")
		if self.change_amount and self.currency != change_account_currency:
			frappe.throw(_("Change Account Currency {} does not match with Invoice Currency {}").format(change_account_currency, self.currency))
```

File: erpnext/accounts/doctype/pos_invoice/pos_invoice.py [hunk 3]
```diff
@@ -593,3 +593,26 @@ def append_payment(payment_mode):
...
+		payment.type = payment_mode.type
+
+	doc.set('payments', [])
+	invalid_modes = []
+	mode_of_payments = [d.mode_of_payment for d in pos_profile.get('payments')]
+	mode_of_payments_info = get_mode_of_payments_info(mode_of_payments, doc.company)
+
+	for row in pos_profile.get('payments'):
+		acct = frappe.db.get_value("Mode of Payment Account", {"parent": row.mode_of_payment, "company": doc.company}, "default_account")
+		if frappe.db.get_value("Account", acct, "account_currency") == doc.currency:
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/pos_invoice/pos_invoice.py:612]:
```suggestion
		mop_account = mode_of_payments_info.get(row.mode_of_payment, {}).get('default_account')
		if mop_account and \
			frappe.db.get_value("Account", mop_account, "account_currency", cache=True) == doc.currency:
```

**Review by nextchamp-saqib — CHANGES_REQUESTED** [review]

**Comment by this-gavagai** [comment]
Thank you so much for reviewing this @nextchamp-saqib! I am travelling for a week or so right now and without easy access to a development environment, but I'll review the code changes and create the tests as soon as I can.

**Pull request closed without merging**